### ISSUE TITLE
unwanted warning message when using `..1` in a purrr loop

### ISSUE THREAD

**Issue by kendonB** [wlandau/targets#164]
## Prework

* [x] I understand and agree to the [code of conduct](https://github.com/wlandau/targets/blob/master/CODE_OF_CONDUCT.md).
* [x] I understand and agree to the [contributing guidelines](https://github.com/wlandau/targets/blob/master/CONTRIBUTING.md).
* [x] I am reasonably sure this is a genuine bug in `targets` and most likely not a user error. (If you run into an error and do not know the cause, please submit a "Trouble" issue instead.)
* [x] Be considerate of the maintainer's time and make it as easy as possible to troubleshoot any problems you identify. **That means posting a runnable, complete, easy to understand [reproducible example](https://www.tidyverse.org/help/) that lets the mainainer fully replicate the issue and see what you see**. Please format your code according to the [tidyverse style guide](https://style.tidyverse.org/).

## Description

I see the below warning message when defining a target that uses a `purrr` numeric counter.

## Reproducible example

``` r
library(targets)
mytarget = tar_target(targetname, {
  list(1) %>% 
    pmap(~..1)
})
#> Warning: <anonymous>: ..1 may be used in an incorrect context
```

<sup>Created on 2020-09-18 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 4.0.2 (2020-06-22)
#>  os       Ubuntu 20.04.1 LTS          
#>  system   x86_64, linux-gnu           
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_NZ.UTF-8                 
#>  ctype    en_NZ.UTF-8                 
#>  tz       Pacific/Auckland            
#>  date     2020-09-18                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version    date       lib source                          
#>  assertthat    0.2.1      2019-03-21 [3] CRAN (R 4.0.0)                  
#>  backports     1.1.10     2020-09-15 [3] CRAN (R 4.0.2)                  
#>  callr         3.4.4      2020-09-07 [3] CRAN (R 4.0.2)                  
#>  cli           2.0.2      2020-02-28 [3] CRAN (R 4.0.0)                  
#>  codetools     0.2-16     2018-12-24 [4] CRAN (R 4.0.0)                  
#>  crayon        1.3.4      2017-09-16 [3] CRAN (R 4.0.0)                  
#>  data.table    1.13.0     2020-07-24 [3] CRAN (R 4.0.2)                  
#>  desc          1.2.0      2018-05-01 [3] CRAN (R 4.0.0)                  
#>  devtools      2.3.1      2020-07-21 [3] CRAN (R 4.0.2)                  
#>  digest        0.6.25     2020-02-23 [3] CRAN (R 4.0.0)                  
#>  ellipsis      0.3.1      2020-05-15 [3] CRAN (R 4.0.0)                  
#>  evaluate      0.14       2019-05-28 [3] CRAN (R 4.0.0)                  
#>  fansi         0.4.1      2020-01-08 [3] CRAN (R 4.0.0)                  
#>  fs            1.5.0      2020-07-31 [3] CRAN (R 4.0.2)                  
#>  glue          1.4.2      2020-08-27 [3] CRAN (R 4.0.2)                  
#>  highr         0.8        2019-03-20 [3] CRAN (R 4.0.0)                  
#>  htmltools     0.5.0      2020-06-16 [3] CRAN (R 4.0.1)                  
#>  igraph        1.2.5      2020-03-19 [3] CRAN (R 4.0.0)                  
#>  knitr         1.29       2020-06-23 [3] CRAN (R 4.0.2)                  
#>  lifecycle     0.2.0      2020-03-06 [3] CRAN (R 4.0.0)                  
#>  magrittr      1.5        2014-11-22 [3] CRAN (R 4.0.0)                  
#>  memoise       1.1.0      2017-04-21 [3] CRAN (R 4.0.0)                  
#>  pillar        1.4.6      2020-07-10 [3] CRAN (R 4.0.2)                  
#>  pkgbuild      1.1.0      2020-07-13 [3] CRAN (R 4.0.2)                  
#>  pkgconfig     2.0.3      2019-09-22 [3] CRAN (R 4.0.0)                  
#>  pkgload       1.1.0      2020-05-29 [3] CRAN (R 4.0.0)                  
#>  prettyunits   1.1.1      2020-01-24 [3] CRAN (R 4.0.0)                  
#>  processx      3.4.4      2020-09-03 [3] CRAN (R 4.0.2)                  
#>  ps            1.3.4      2020-08-11 [3] CRAN (R 4.0.2)                  
#>  purrr         0.3.4      2020-04-17 [3] CRAN (R 4.0.0)                  
#>  R6            2.4.1      2019-11-12 [3] CRAN (R 4.0.0)                  
#>  remotes       2.2.0      2020-07-21 [3] CRAN (R 4.0.2)                  
#>  rlang         0.4.7      2020-07-09 [3] CRAN (R 4.0.2)                  
#>  rmarkdown     2.3        2020-06-18 [3] CRAN (R 4.0.1)                  
#>  rprojroot     1.3-2      2018-01-03 [3] CRAN (R 4.0.0)                  
#>  sessioninfo   1.1.1      2018-11-05 [3] CRAN (R 4.0.0)                  
#>  stringi       1.5.3      2020-09-09 [3] CRAN (R 4.0.2)                  
#>  stringr       1.4.0      2019-02-10 [3] CRAN (R 4.0.0)                  
#>  targets     * 0.0.0.9000 2020-09-18 [1] Github (wlandau/targets@28285f1)
#>  testthat      2.3.2      2020-03-02 [3] CRAN (R 4.0.0)                  
#>  tibble        3.0.3      2020-07-10 [3] CRAN (R 4.0.2)                  
#>  tidyselect    1.1.0      2020-05-11 [3] CRAN (R 4.0.0)                  
#>  usethis       1.6.1      2020-04-29 [3] CRAN (R 4.0.0)                  
#>  vctrs         0.3.4      2020-08-29 [3] CRAN (R 4.0.2)                  
#>  withr         2.2.0      2020-04-20 [3] CRAN (R 4.0.0)                  
#>  xfun          0.17       2020-09-09 [3] CRAN (R 4.0.2)                  
#>  yaml          2.2.1      2020-02-01 [3] CRAN (R 4.0.0)                  
#> 
#> [1] /home/kendonb/R/x86_64-pc-linux-gnu-library/4.0
#> [2] /usr/local/lib/R/site-library
#> [3] /usr/lib/R/site-library
#> [4] /usr/lib/R/library
```

</details>

## Expected result

Same but no warning

## Diagnostic information

Done above

**Comment by wlandau** [comment]
I know it's annoying but, I don't think anything can be done. In order to solve #143 and detect functions called from `purrr` lambda syntax, `targets` needs to manually substitute `~` for something else before calling `codetools::findGlobals()`.

https://github.com/wlandau/targets/blob/28285f19fb0b9158f23e969b7be56dc66a9ed285/R/utils_language.R#L1-L5

The warning from `codetools` appears [here](https://github.com/cran/codetools/blob/e10d13c12831c8cb869c7564a058eb183a884245/R/codetools.R#L416) because `findGlobals()` detects the symbol `..1` using [`isDDsym()`](https://github.com/cran/codetools/blob/e10d13c12831c8cb869c7564a058eb183a884245/R/codetools.R#L1170-L1173). `codetools` never reaches that part of the logic for formulas.

In other words, because we need to replace `~`, we are deliberately taking all the symbols in the formula out of context. So the observation in the `codetools` warning is actually correct, despite the fact that we have a good reason for triggering it.

**Comment by kendonB** [comment]
What do you recommend users to do to avoid seeing this confusing warning?

**Comment by wlandau** [comment]
`suppressWarnings()`? Other workarounds seem extreme for such a harmless warning.
